tests/infra_test.go

```diff
@@ -1054,6 +1054,78 @@ var _ = Describe("[Serial][sig-compute]Infrastructure", func() {
 		})
 
 		Context("basic labelling", func() {
+			It("skip node reconciliation when node has skip annotation", func() {
+				nonExistingCPUModelLabel := v1.CPUModelLabel + "someNonExistingCPUModel"
+				type patch struct {
+					Op    string            `json:"op"`
+					Path  string            `json:"path"`
+					Value map[string]string `json:"value"`
+				}
+				for i, node := range nodesWithKVM {
+					node.Labels[nonExistingCPUModelLabel] = "true"
+					p := []patch{
+						{
+							Op:    "add",
+							Path:  "/metadata/labels",
+							Value: node.Labels,
+						},
+					}
+					if i == 0 {
+						node.Annotations[v1.LabellerSkipNodeAnnotation] = "true"
+
+						p = append(p, patch{
+							Op:    "add",
+							Path:  "/metadata/annotations",
+							Value: node.Annotations,
+						})
+					}
+					payloadBytes, err := json.Marshal(p)
+					Expect(err).ToNot(HaveOccurred())
+
+					_, err = virtClient.CoreV1().Nodes().Patch(context.Background(), node.Name, types.JSONPatchType, payloadBytes, metav1.PatchOptions{})
+					Expect(err).ToNot(HaveOccurred())
+				}
+
+				nodesWithKVM = tests.GetNodesWithKVM()
+
+				for _, node := range nodesWithKVM {
+					_, skipAnnotationFound := node.Annotations[v1.LabellerSkipNodeAnnotation]
+					_, customLabelFound := node.Labels[nonExistingCPUModelLabel]
+
+					if customLabelFound && !skipAnnotationFound {
+						Expect(customLabelFound && !skipAnnotationFound).To(Equal(true), "node with user defined label should contain skip annotation")
+					}
```

Comment:
i'm not sure if this makes sense. 

`Expect(customLabelFound && !skipAnnotationFound).To(Equal(true)` is wrapped in an if statement where `customLabelFound && !skipAnnotationFound`

so... this can never fail. The check doesn't look accurate. 

If skipAnnotationFound then check to verify the custom label exists

if !skipAnnotationFound then check to verify the custom label is removed.